bin/menmosd/src/menmosd/node/test/mock.rs

```diff
@@ -21,14 +38,56 @@ fn tag_to_kv(tag: &str) -> Result<(&str, &str)> {
 }
 
 #[derive(Default)]
-pub struct MockDocIdMap {
+struct MockUserStore {
+    users: Mutex<HashMap<String, String>>,
+}
+
+#[async_trait]
+impl Flush for MockUserStore {
+    async fn flush(&self) -> Result<()> {
+        Ok(())
+    }
+}
+
+impl UserStore for MockUserStore {
+    fn authenticate(&self, username: &str, password: &str) -> Result<bool> {
+        let guard = self.users.lock().unwrap();
+        Ok(guard.get(username).cloned().unwrap_or(String::default()) == password)
+    }
+
+    fn add_user(&self, username: &str, password: &str) -> Result<()> {
+        let mut guard = self.users.lock().unwrap();
+        guard.insert(username.to_string(), password.to_string());
+        Ok(())
+    }
+
+    fn has_user(&self, username: &str) -> Result<bool> {
+        let guard = self.users.lock().unwrap();
+        Ok(guard.contains_key(username))
+    }
+
+    fn iter(&self) -> DynIter<'static, Result<String>> {
+        // Returning an iterator on something protected by a mutex = cursed.
```

Comment:
Wouldn't that be worthy of a TODO?

Reply:
Eh not really, the method is just there so the test compiles, we don't use it so I don't really see the value.